test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,226 @@
+			'aria-current',
+			'true'
+		);
+		await expect( variations.nth( 1 ) ).toHaveAttribute(
+			'aria-current',
+			'false'
+		);
+		await expect( variations.nth( 2 ) ).toHaveAttribute(
+			'aria-current',
+			'false'
+		);
+	} );
+
+	test( 'should apply preset colors and font sizes in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.browseStyles();
+		await page.click( 'role=button[name="pink"i]' );
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Colors styles"i]' );
+
+		await expect(
+			page.locator(
+				'role=button[name="Colors background styles"i] >> data-testid=background-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator(
+				'role=button[name="Colors text styles"i] >> data-testid=text-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(74, 7, 74\)/ );
+
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Typography styles"i]' );
+		await page.click( 'role=button[name="Typography Text styles"i]' );
+
+		await expect(
+			page.locator( 'css=.components-font-size-picker__header__hint' )
+		).toHaveText( 'Medium(px)' );
+	} );
+
+	test( 'should apply custom colors and font sizes in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.browseStyles();
+		await page.click( 'role=button[name="yellow"i]' );
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Colors styles"i]' );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] [data-testid="background-color-indicator"]'
+			)
+		).toHaveCSS( 'background', /rgb\(255, 239, 11\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] [data-testid="text-color-indicator"]'
+			)
+		).toHaveCSS( 'background', /rgb\(25, 25, 17\)/ );
+
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Typography styles"i]' );
+		await page.click( 'role=button[name="Typography Text styles"i]' );
+
+		// TODO: to avoid use classnames to locate these elements,
+		//  we could provide accessible attributes to the source code in packages/components/src/font-size-picker/index.js.
+		await expect(
+			page.locator( 'css=.components-font-size-picker__header__hint' )
+		).toHaveText( '(Custom)' );
+
+		await expect(
+			page.locator(
+				'css=.components-font-size-picker input[aria-label="Custom"]'
```

Comment:
Don't we already have accessible role and name for this element?

Reply:
Sorry, I forgot about 'spinbutton'

All I see are green ✅  :trollface: